src/core/namespace.c

```diff
@@ -464,6 +464,52 @@ static int append_mount_images(MountList *ml, const MountImage *mount_images, si
         return 0;
 }
 
+static void verities_done_many(VeritySettings *v, size_t n) {
+        assert(v || n == 0);
+
+        FOREACH_ARRAY(i, v, n)
+                verity_settings_done(i);
+
+        free(v);
+}
+
+static int verities_find(const char *path, VeritySettings **verities, size_t *n_verities) {
+        _cleanup_(verity_settings_done) VeritySettings verity = VERITY_SETTINGS_DEFAULT;
+        bool dup = false;
+        int r;
+
+        assert(path);
+        assert(verities);
+        assert(n_verities);
+
+        r = verity_settings_load(&verity, path, /* root_hash_path= */ NULL, /* root_hash_sig_path= */ NULL);
```

Comment:
so you load this now, and then throw it away, and don't actually use it?

Reply:
it's appended to the verities array (if missing), which is a mutable object

`(*verities)[(*n_verities)++] = TAKE_GENERIC(verity, VeritySettings, VERITY_SETTINGS_DEFAULT);`

Reply:
yeah, which nothing uses though except for the dup detection. so it's basically thrown away, and then loaded again later.

Reply:
Yes, this is an input check. it's not mounting or changing any state, just reading info, so I don't think it's a problem

Reply:
if something "is a problem" is not the benchmark. The benchmark is higher than that, i.e. if this is somewhat clean code. And no, loading this once and throwing it away is not clean.